streams/src/test/java/org/apache/kafka/streams/integration/FanoutIntegrationTest.java

```diff
@@ -0,0 +1,166 @@
+        cluster.createTopic(OUTPUT_TOPIC_C);
+    }
+
+    @Test
+    public void shouldFanoutTheInput() throws Exception {
+        List<String> inputValues = Arrays.asList("Hello", "World");
+        List<String> expectedValuesForB = new ArrayList<>();
+        List<String> expectedValuesForC = new ArrayList<>();
+        for (String input : inputValues) {
+            expectedValuesForB.add(input.toUpperCase(Locale.getDefault()));
+            expectedValuesForC.add(input.toLowerCase(Locale.getDefault()));
+        }
+
+        //
+        // Step 1: Configure and start the processor topology.
+        //
+        KStreamBuilder builder = new KStreamBuilder();
+
+        Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "fanout-integration-test");
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG, cluster.zKConnectString());
+        streamsConfiguration.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+
+        KStream<byte[], String> stream1 = builder.stream(INPUT_TOPIC_A);
+        KStream<byte[], String> stream2 = stream1.mapValues(
+            new ValueMapper<String, String>() {
+                @Override
+                public String apply(String value) {
+                    return value.toUpperCase(Locale.getDefault());
+                }
+            });
+        KStream<byte[], String> stream3 = stream1.mapValues(
+            new ValueMapper<String, String>() {
+                @Override
+                public String apply(String value) {
+                    return value.toLowerCase(Locale.getDefault());
+                }
+            });
+        stream2.to(OUTPUT_TOPIC_B);
+        stream3.to(OUTPUT_TOPIC_C);
+
+        KafkaStreams streams = new KafkaStreams(builder, streamsConfiguration);
+        streams.start();
+
+        // Wait briefly for the topology to be fully up and running (otherwise it might miss some or all
+        // of the input data we produce below).
+        Thread.sleep(5000);
+
+        //
+        // Step 2: Produce some input data to the input topic.
+        //
+        Properties producerConfig = new Properties();
+        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.bootstrapServers());
+        producerConfig.put(ProducerConfig.ACKS_CONFIG, "all");
+        producerConfig.put(ProducerConfig.RETRIES_CONFIG, 0);
+        producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class);
+        producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        IntegrationTestUtils.produceValuesSynchronously(INPUT_TOPIC_A, inputValues, producerConfig);
+
+        // Give the stream processing application some time to do its work.
+        Thread.sleep(10000);
```

Comment:
Right now I think we should keep the (high) sleep times as is.

But in general we may want to lower the sleep times in this PR.  I don't know what a "good" value would be (preferably we'd be able to use `TestUtils.waitUntilTrue` eventually) -- using sleep just s*cks -- but I think Jenkins should be running a bit faster than Travis.  Perhaps we should give it a try with `5s` rather than `10s`?

That said, we perhaps don't want another iteration on this code, given that the 0.10 vote already started.